lib/src/git.rs

```diff
@@ -1452,9 +1480,15 @@ impl<'a> GitFetch<'a> {
                          * because `to_glob()` escapes such `*`s as `[*]`. */
                         |glob| !glob.contains(INVALID_REFSPEC_CHARS),
                     )
-                    .map(|glob| format!("+refs/heads/{glob}:refs/remotes/{remote_name}/{glob}"))
+                    .map(|glob| {
+                        RefSpec::forced(
+                            format!("refs/heads/{glob}"),
+                            format!("refs/remotes/{remote_name}/{glob}"),
+                        )
+                    })
             })
-            .collect::<Option<_>>()
+            .map(|refspec| refspec.map(|r| r.to_git_format()))
+            .collect::<Option<Vec<_>>>()
```

Comment:
nit: `Vec<_>` annotation is redundant since the bound variable has type annotation.